model/riscv_pmp_regs.sail

```diff
@@ -154,15 +163,34 @@ function pmpReadCfgReg(n) = {
 }
 
 /* Helpers to handle locked entries */
+function pmpLockBit(cfg: Pmpcfg_ent) -> bool =
+  cfg.L() == 0b1
+   
 function pmpLocked(cfg: Pmpcfg_ent) -> bool =
-   cfg.L() == 0b1
+  (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0)
 
 function pmpTORLocked(cfg: Pmpcfg_ent) -> bool =
-   (cfg.L() == 0b1) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
-
+  (cfg.L() == 0b1) & (mseccfg.RLB() == 0b0) & (pmpAddrMatchType_of_bits(cfg.A()) == TOR)
+  
+/* Function to check whether executable region can be added or not in accordance to 4b of Smepmp spec */
+function checkExe(cfg: Pmpcfg_ent,v: bits(8)) -> Pmpcfg_ent =
+  if ((mseccfg.MML() == 0b1) & (mseccfg.RLB() == 0b0) & (v[7] == bitone)) then {
+   if ((v[0] == bitzero & v[1] == bitzero & v[2] == bitone) |
```

Comment:
Please construct the new Pmpcfg_ent *first* and then legalise it based on its named fields rather than do all this opaque bit indexing

Reply:
(See legalize_mstatus, for example)

Reply:
Removed the opaque bit indexing by constructing 4 legalized Pmpcfg_ent and then made the comparison with the configuration to be written and made the decision based on that.